weaver/tranform/transform.py

```diff
@@ -0,0 +1,379 @@
+            image_w, image_h = im.size
+
+            if image_w > image_h:
+                new_pdf.add_page(orientation='L')
+                _w = pdf_height
+                _h = pdf_width
+            else:
+                new_pdf.add_page(orientation='P')
+                _h = pdf_height
+                _w = pdf_width
+
+            while image_h > _h or image_w > _w:
+                image_h *= 0.999
+                image_w *= 0.999
+
+            new_pdf.image(im, x=(_w - image_w) / 2, y=(_h - image_h) / 2, w=image_w, h=image_h)
+
+    new_pdf.output(o, 'F')
+
+        #ims[0].save(o, save_all=True, append_images=ims)
+
+
+@exception_handler
+def csv_to_json(i, o):
+    with open(i, encoding='utf-8') as csvf:
+        csvReader = csv.DictReader(csvf)
+
+        for i in range(len(csvReader.fieldnames)):
+            if csvReader.fieldnames[i] == "":
+                csvReader.fieldnames[i] = "unknown_" + str(i)
+
+        ret = []
+        for rows in csvReader:
+            ret.append({"data": rows})
+        # datas = {"datas": ret}
+        write_content(o, {"datas": ret})
+
+
+@exception_handler
+def csv_to_xml(i, o):
+    p = i + ".json"
+    csv_to_json(i, p)
+    data = readfromjson(p)
+    write_content(o, json2xml.Json2xml(data, item_wrap=False).to_xml())
+
+
+@exception_handler
+def json_to_xml(i, o):
+    data = readfromjson(i)
+    write_content(o, json2xml.Json2xml(data, item_wrap=False).to_xml())
+
+
+@exception_handler
+def json_to_yaml(i, o):
+    with open(i, 'r') as file: configuration = json.load(file)
+    with open(o, 'w') as yaml_file: yaml.dump(configuration, yaml_file)
+
+
+@exception_handler
+def yaml_to_json(i, o):
+    with open(i, 'r') as file: configuration = yaml.safe_load(file)
+    with open(o, 'w') as json_file: json.dump(configuration, json_file)
+
+
+@exception_handler
+def json_to_csv(i, o):
+    with open(i, encoding='utf-8') as inputfile:
+        df = pd.read_json(inputfile)
+        df.to_csv(o, encoding='utf-8', index=False)
+
+
+@exception_handler
+def xml_to_json(i, o):
+    write_content(o, xmltodict.parse(get_content(i)))
+
+
+@exception_handler
+def html_to_txt(i, o):
+    write_content(o, ' '.join(BeautifulSoup(get_content(i), "html.parser").stripped_strings))
+
+
+@exception_handler
+def yaml_to_csv(i, o):
+    yaml_to_json(i, i + ".json")
+    json_to_csv(i + ".json", o)
+
+
+@exception_handler
+def yaml_to_xml(i, o):
+    yaml_to_json(i, i + ".json")
+    json_to_xml(i + ".json", o)
+
+
+@exception_handler
+def xml_to_yaml(i, o):
+    xml_to_json(i, i + ".json")
+    json_to_yaml(i + ".json", o)
+
+
+@exception_handler
+def csv_to_yaml(i, o):
+    csv_to_json(i, i + ".json")
+    json_to_yaml(i + ".json", o)
+
+
+class Transform:
+    def __init__(self, file_path, current_media_type: str, wanted_media_type: str):
+        self.file_path = file_path
+        self.cmt = current_media_type.lower()
+        self.wmt = wanted_media_type.lower()
+        self.output_path = self.file_path
```

Comment:
The output path needs to be modifiable (either from init arg, or modified after before the `process`/`get` calls.
More importantly, no additional files should be added under the original WPS outputs sub-directories, because their contents could be critical to the packing/detection of other apps in a workflow chain. 
We need to consider the creation of some "sibling" directory next to the WPS-output-ID dir to avoid modifying the output checksums.